### PR TITLE
add ner manual annotator-training recipe

### PR DESCRIPTION
**Pull request by kabirkhan**
We want a recipe for helping to train annotators for NER

### CODE REVIEW

**Comment by magdaaniol** [comment]
The idea I had here was to give a bit more elaborate error messages that, minimally, would point the user to the relevant section in the guidelines or, better, provide example-specific explanations. I can think of two ways how to achieve that: 1) provide another json as input to the recipe that would contain the explanations specific to each example 2) store explanations directly in the golden dataset under `meta` or another key. WDYT?

**Comment by magdaaniol** [comment]
As for customizing the alert popup, I also need this for another recipe so I'll dig a little / ask around whether it can be achieved easily.

**Comment by kabirkhan** [comment]
So the UI code just calls `alert` directly here with a message from the Prodigy API. https://github.com/explosion/prodigy/blob/48636e54e5f38479c8801ee1aa94d2b8affb03cc/app/containers/App.jsx#L311

We do have a `Modal` component for displaying the instructions so we could adapt that if we'd like.

File: ner/ner_manual_annotator_training.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    stream = get_stream(
+        source,
+        loader=loader,
+        rehash=True,
+        dedup=True,
+        input_key="text",
+        is_binary=False,
+    )
+    known_answers = list(get_stream(
+        source,
```
Comment (magdaaniol) [inline ner/ner_manual_annotator_training.py:105]: this should be `known_answers` right?

File: ner/ner_manual_annotator_training.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+
+def ensure_span_text(eg: TaskType) -> TaskType:
+    for span in eg["spans"]:
+        if "text" not in span:
+            span["text"] = eg["text"][span["start"]:span["end"]]
+    return eg
+
+
+def validate_answer(answer: TaskType, *, known_answers: List[TaskType]):
```
Comment (magdaaniol) [inline ner/ner_manual_annotator_training.py:27]: I think we should also check the label and provide some custom explanations in respect. I think the `known_answers` json could have this field in the `meta`. I'll code up a proposal on a separate branch.

**Pull request closed without merging**